File: Majority-Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    const map = new Map();

    for (const num of nums) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    for (const [num, count] of map) {
        if (count > nums.length / 2) {
            return num;
        }
    }

    return 0;  
};

File: Majority-Element.py
class Solution(object):
    def majorityElement(self, nums):
        \\\
        :type nums: List[int]
        :rtype: int
        \\\
        n = len(nums)
        count_map = defaultdict(int)

        for num in nums:
            count_map[num] += 1

        for num, count in count_map.items():
            if count > n // 2:
                return num

        return 0  
        